### PR TITLE
task/SWTCH-2036 siwe

### PR DESCRIPTION
**Pull request by JGiter**
### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or update existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: src/modules/cache-client/cache-client.service.ts [hunk 1]
```diff
@@ -85,16 +86,10 @@ export class CacheClient implements ICacheClient {
     // If refresh token failed or access token is not valid, then sign new identity token
     if (!(await this.isAuthenticated())) {
       getLogger().info('[CACHE CLIENT] obtaining new tokens');
-      delete this._httpClient.defaults.headers.common['Authorization'];
```
Comment (jrhender) [inline src/modules/cache-client/cache-client.service.ts:88]: Is this line still needed? It seems to be unrelated to the authentication protocol change

**Review by jrhender — COMMENTED** [review]
This is looking good 👍

File: src/modules/cache-client/cache-client.service.ts (hunk 1 above)
Comment (JGiter) [reply src/modules/cache-client/cache-client.service.ts:88]: I removed it, because this line do nothing. Next step in this method signs in to ssi-hub, which doesn't need Authorization header and then Authrization header is assigned new value. But of course we can remove this in separate PR

File: jest.config.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ module.exports = {
     },
     transformIgnorePatterns: ["node_modules/!(@energyweb/ekc)", "node_modules/.+\\.!(mjs)$"],
     testEnvironment: "node",
-    testRegex: "(/(e2e|src)/(.|\\.)*\\.(e2e|spec)\\.ts$)",
+    testRegex: "(/(e2e|src)/(.|\\.)*auth\\.(e2e|spec)\\.ts$)",
```
Comment (jrhender) [inline jest.config.js:9]: I think we should remove this before merging, right?

File: src/config/cache.config.ts [hunk 3]
```diff
@@ -1,14 +1,18 @@
 import { EWC_CHAIN_ID, VOLTA_CHAIN_ID } from '@energyweb/credential-governance';
 import { CacheServerClientOptions } from '../modules/cache-client/cache-client.types';
 
-const cacheConfig: Record<number, CacheServerClientOptions> = {
+const defaultConfig: Record<number, CacheServerClientOptions> = {
   [VOLTA_CHAIN_ID]: {
     url: 'https://identitycache-staging.energyweb.org/v1/',
```
Comment (jrhender) [inline src/config/cache.config.ts:6]: If default domain is `switchboard-dev`, we should change `url` to `identitycache-dev`. I think this is actually a good time to do this as this is already a breaking change due to support for the previous login method being removed.

```suggestion
    url: 'https://identitycache-dev.energyweb.org/v1/',
```

File: src/modules/cache-client/auth.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+   * Signs in to ssi-hub. Authentication method configurations are looked for in `cacheConfigs`
+   *
+   * @returns authentication tokens
+   */
+  async signIn(): Promise<AxiosResponse<AuthTokens>> {
+    const {
+      data: { nonce },
+    } = await this.http.post<{ nonce: string }>('/login/siwe/initiate');
+    const siweMessage = new SiweMessage({
+      statement: `I accept the Terms of Service: ${this.config.domain}`,
```
Comment (jrhender) [inline src/modules/cache-client/auth.ts:33]: I would remove the `statement` for now, unless it is necessary.
```suggestion
```

File: src/modules/cache-client/auth.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      nonce,
+      domain: new URL(this.config.domain).host,
+      address: this.signerService.address,
+      uri: `${cacheConfigs()[this.signerService.chainId].url}login/siwe/verify`,
```
Comment (jrhender) [inline src/modules/cache-client/auth.ts:37]: I think this is potentially a bit brittle as it will have a missing`/` if `url` doesn't end with a `/`. I'm not sure what the better approach is though. Maybe using something like this? 
```suggestion
      const url = cacheConfigs()[this.signerService.chainId].url
      uri: `${url}${url.substring(url.length - 1) === '/' ? '' : '/'} login/siwe/verify`,
```

File: src/config/cache.config.ts (hunk 3 above)
Comment (jrhender) [reply src/config/cache.config.ts]: @JGiter could you also note the above in the migration guide documentation please? If folks had previously relied on the default value, they will need to specify `'https://identitycache-staging.energyweb.org/v1/' as the default has been changed to 'https://identitycache-dev.energyweb.org/v1/'

Comment (JGiter) [reply src/config/cache.config.ts]: Good idea

File: src/modules/cache-client/auth.ts (hunk 4 above)
Comment (JGiter) [reply src/modules/cache-client/auth.ts]: Ok

File: src/modules/cache-client/auth.ts (hunk 5 above)
Comment (JGiter) [reply src/modules/cache-client/auth.ts]: Yes, it is a bit dangerous. I changed it to new URL('/login/siwe/verify', cacheConfigs()[this.signerService.chainId].url).href. This will create correct uri regardless of trailing slash

**Review by nichonien — APPROVED** [review]
Thanks @JGiter , great work!

**Comment by jrhender** [comment]
@JGiter Can we move https://github.com/energywebfoundation/iam-client-lib/pull/678/commits/e004d1a1f4291650297e9e0d3397e6a1d26df06a to a separate PR?
I think the PR as it was prior to the commit was mergeable (IMO) and the changes introduced by the commit might require further discussion

**Comment by JGiter** [comment]
> @JGiter Can we move [e004d1a](https://github.com/energywebfoundation/iam-client-lib/commit/e004d1a1f4291650297e9e0d3397e6a1d26df06a) to a separate PR? I think the PR as it was prior to the commit was mergeable (IMO) and the changes introduced by the commit might require further discussion

Ok, I moved this change to another branch if we decide to proceed with such approach https://github.com/energywebfoundation/iam-client-lib/tree/sign_non_eip-191
